src/vstest.console/Processors/CollectArgumentProcessor.cs

```diff
@@ -182,11 +182,47 @@ internal static void AddDataCollectorToRunSettings(string argument, IRunSettings
             EnableDataCollectorUsingFriendlyName(argument, dataCollectionRunSettings);
 
             runSettingsManager.UpdateRunSettingsNodeInnerXml(Constants.DataCollectionRunSettingsName, dataCollectionRunSettings.ToXml().InnerXml);
+
+            // Add inproc data collector to runsetings if coverlet code coverage is enabled
+            if (string.Equals(argument, Constants.CoverletDataCollectorFriendlyName, StringComparison.OrdinalIgnoreCase))
+            {
+                runSettingsManager.UpdateRunSettingsNodeInnerXml(Constants.InProcDataCollectionRunSettingsName, GetInprocDataCollectorSettings().InnerXml);
```

Comment:
This will replace all existing inproc data collectors. Need to modify the code.
To use inprocdatacolletion, use something like `var inprocDataCollectionRunSettings = XmlRunSettingsUtilities.GetInProcDataCollectionRunSettings(settings) ?? new DataCollectionRunSettings();`

Reply:
Done